crates/bws/src/main.rs

```diff
@@ -245,6 +250,18 @@ async fn process_commands() -> Result<()> {
         return Ok(());
     };
 
+    if let Commands::Completions { shell } = command {
+        let Some(shell) = shell.or_else(Shell::from_env) else {
+            eprintln!("Couldn't autodetect a valid shell. Run `bws completions --help` for more info.");
+            std::process::exit(1);
+        };
+
+        let mut cmd = Cli::command();
+        let name = cmd.get_name().to_string();
+        clap_complete::generate(shell, &mut cmd, name, &mut std::io::stdout());
+        return Ok(());
+    }
```

Comment:
We're starting to have a few if let here, should we try and use a switch?

Reply:
That's a good point, we should probably consider extracting them to a separate function in the future because the process_commands is already getting pretty big, but that's a bigger refactoring.